### ISSUE TITLE
Bug: NVDA reads element as "clickable" because onClick is not removed (but replaced with noop())

### ISSUE THREAD

**Issue by NullPainter2** [facebook/react#21501]
**Fact:** 

NVDA, and possibly other screen readers, read element as "clickable ..." when element has onClick event handler set.

**Problem:** 

React 16.13.1 replaces onClick handler with a `noop()` when onClick becomes either undefined or unset, but this still makes screen reader read "clickable ..."

**Reproduction:**

-  JSX should have a conditional onClick like this 
```
<div tabindex="0" {... isMobile && {onClick: () => { some code }}> ...
```
or this
```
<div tabindex="0" {... isMobile ? {onClick: () => { some code } : {onClick: undefined}}> ...
```
- change screen size from mobile to desktop
- NVDA will read element

**Comment by eps1lon** [comment]
Duplicate of https://github.com/facebook/react/issues/20895

**Comment by NullPainter2** [comment]
@eps1lon

I have React 16.x (but 17.x has same error as 16.x). 

I have different problem than OP from duplicate  #20895 . 

My problem is not with **first** element inside of react root which NVDA reports as "clickable" (which OP has), but with element which is deep down into the React tree. Also when I do not specify `onClick()`, then NVDA does not read element as "clickable". NVDA reads element as "clickable" when React adds `noop()` **onto** the element (which happens after resize ...).

Also from MDN article which OP linked I quote _"An element is clickable if it has an onclick event handler defined."_ . This article from Mozilla follows W3C standard for web accessibility, so React does not follow web standard here.

Please reopen my bug. :)
